relay-server/src/utils/dynamic_sampling.rs

```diff
@@ -1,152 +1,228 @@
 //! Functionality for calculating if a trace should be processed or dropped.
 //!
 use std::convert::TryInto;
-use std::fmt;
 
 use actix::prelude::*;
 use futures::{future, prelude::*};
 use rand::{distributions::Uniform, Rng};
 use rand_pcg::Pcg32;
-use serde::de::Visitor;
-use serde::{Deserialize, Deserializer, Serialize};
+use serde::{Deserialize, Serialize};
 
-use relay_common::{ProjectId, ProjectKey, Uuid};
+use relay_common::{EventType, ProjectKey, Uuid};
 use relay_filter::GlobPatterns;
 use relay_general::protocol::{Event, EventId};
 
 use crate::actors::project::{GetCachedProjectState, GetProjectState, Project, ProjectState};
 use crate::envelope::{Envelope, ItemType};
 
-#[derive(Debug, Clone, Copy, Serialize, Deserialize, PartialEq, Eq)]
+#[derive(Debug, Copy, Clone, Serialize, Deserialize, Eq, PartialEq)]
 #[serde(rename_all = "camelCase")]
-pub enum SamplingStrategy {
-    Trace, // Rules that apply to Transaction items
-    Event, // Rules that apply to Event items
+pub enum RuleType {
+    /// A trace rule applies only to transactions and it is applied on the trace info
+    Trace,
+    /// A transaction rule applies to transactions and it is applied  on the transaction event
+    Transaction,
+    /// A non transaction rule applies to Errors, Security events...every type of event that
+    /// is not a Transaction
+    Error,
+}
+
+/// The value of a field extracted from a FieldValueProvider (Event or TraceContext)
+#[derive(Debug, Clone)]
+pub enum FieldValue<'a> {
+    String(&'a str),
+    None,
 }
 
-/// A sampling rule defined by user in Organization options.
 #[derive(Debug, Clone, Serialize, Deserialize)]
 #[serde(rename_all = "camelCase")]
-pub struct SamplingRule {
-    /// The project ids for which the sample rule applies, empty applies to all
-    #[serde(default)]
-    pub project_ids: Vec<ProjectId>,
-    /// A set of Glob patterns for which the rule applies, empty applies to all
-    #[serde(default)]
-    pub releases: GlobPatterns,
-    /// A set of user_segments for which the rule applies, empty applies to all
+pub struct EqCondData {
```

Comment:
Please add doc strings to all public types and fields.

_nit_: The `Data` suffix here seems redundant, and some other types are named with the full "Condition". How about we go with: `EqCondition`? 

```suggestion
pub struct EqCondition {
```